src/shells.ts

```diff
@@ -0,0 +1,30 @@
+import * as vscode from 'vscode';
+
+export enum OS {
+  WINDOWS = 1,
+  LINUX = 2
+}
+
+const windowsDefaultShell = 'powershell.exe';
+const linuxDefaultShell = '/bin/bash';
+
+
+export function getShell(os: OS) {
+  const integratedShell = vscode.workspace.getConfiguration('terminal.integrated.shell');
+
+  if (os === OS.WINDOWS) {
+    const windowsShell = integratedShell?.get<string>('windows');
+    if (windowsShell) {
+      return windowsShell;
+    }
+    return windowsDefaultShell;
+  }
+
+  if (os === OS.LINUX) {
+    const linuxShell = integratedShell?.get<string>('linux');
+    if (linuxShell) {
+      return linuxShell;
+    }
+    return linuxDefaultShell;
+  }
```

Comment:
This also connects to mac hosts which would get undefined as the shell currently. I'm not sure there's a way to easily detect this. Sticking with `$SHELL` seems like the most reliable option given this, that should also be a better default thank `/bin/bash`.

Reply:
i think you are right. maybe we should try to use this 
```
if [ "$(uname -s)" == "Darwin" ]; then exec mac_shell; else exec linux_shell; fi
```

Reply:
Good idea :+1:, that seems to be the best of both worlds. We could craft the command depending on the value of the settings, eg if shell.linux is null:

```sh
if [[ "$(uname -s)" = "Darwin" ]]; then macshell; fi || exec $SHELL -l
```

Reply:
shouldn't we use the defaults that vscode use? 
![image](https://user-images.githubusercontent.com/20097801/84381163-84f69d00-abe0-11ea-906a-4d43b1af62dc.png)

getConfiguration(...).get("linux") does not return the defaults thats why i return the default when the shell is null.
if you open a integrated terminal in vscode it opens with the defaults (powershell.exe for windows, /bin/bash for linux and osx).

Also i think we should make this file independent so we can use it on wsl in the future.

Reply:
That default in the setting description is dynamic and is evaluated on the particular machine on startup. The typical case is that it just pulls from `$SHELL`. Here's where the defaults are set: https://github.com/microsoft/vscode/blob/145d6bf352b6cc078c201140a963dbe04b001b57/src/vs/workbench/contrib/terminal/node/terminal.ts#L34-L62

I think you're right with Windows and I'm overthinking it with pwsh since powershell.exe is the default terminal, let's use terminal.integrated.shell.windows if it's set, otherwise powershell.exe.

Reply:
well seen i will change that too ;)